crates/uv-workspace/src/workspace.rs

```diff
@@ -0,0 +1,69 @@
+use std::path::{Path, PathBuf};
+
+use uv_fs::Simplified;
+
+use crate::{Options, PyProjectToml};
+
+/// Represents a project workspace that contains a set of options and a root path.
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Workspace {
+    options: Options,
+    root: PathBuf,
+}
+
+impl Workspace {
```

Comment:
I'm not sure if i'd call that workspace, i'm thinking about having settings, things that are defined e.g. on the user-level, the project level or the cli level which e.g. define indexes, plus having a workspace, an abstraction over a set of packages that are part of single project. That's the main divergence from ruff, in uv (similar to cargo) you're always in a single workspace and/or a single package; i still have to flesh out how to integrate this into uv.

In cargo, you're technically always in a rust workspace, even if you only define a single crate, that's a single crate workspace. Then there's the concept of [configuration](https://doc.rust-lang.org/cargo/reference/config.html) with hierarchical structure and cli overrides.

Reply:
I can call this `configuration` instead?

Reply:
I've been intending to call it a "workspace" whether there's one package or many in your project. I don't know if we need to introduce a separate term?

Reply:
I like `configuration`